validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/Eth1DataCache.java

```diff
@@ -55,6 +55,7 @@ public Eth1DataCache(EventBus eventBus, TimeProvider timeProvider) {
   public void startBeaconChainMode(BeaconState genesisState) {
     this.genesisTime = Optional.of(genesisState.getGenesis_time());
```

Comment:
Should `genesisTime` be `volatile`?

Reply:
Yes, one thread sets it, and potentially two different threads access it, possibly at the same time, so it should be volatile.